lib/appRelationsDiscovery.js

```diff
@@ -0,0 +1,65 @@
+const {validate} = require('./utils');
+
+const supportedSyntax = ['default','mermaid'];
+
+// eslint-disable-next-line sonarjs/cognitive-complexity
+async function getRelations(asyncApiDocs,requestedSyntax = 'default') {
+  if (typeof asyncApiDocs !== 'object') throw new Error('You must pass an array of AsyncApiDocuments on which you wish to discover the relations between');
+  if (!supportedSyntax.includes(requestedSyntax)) return;
+
+  const parsedAsyncApiDocs = await validate(asyncApiDocs);
+  const metrics = new Map();
+  
+  parsedAsyncApiDocs.forEach(doc => {
+    if (doc.hasServers()) {
+      const servers = doc.servers();
+
+      for (const [, credentials] of Object.entries(servers)) {
+        const slug = `${credentials.url()},${credentials.protocol()}`;
+        let relation;
+        if (metrics.has(slug)) {
+          relation = metrics.get(slug);
+        } else {
+          relation = new Map();
+        }
+        // metrics
+        // ServerMap(n-servers) => {
+        //   <Url+protocol> => ChannelMap(n-channels) => {
+        //      <channelName> => { sub: [] , pub: [] }
+        //    },
+        //    ....
+        // }
```

Comment:
Instead of needing comments to explain the structure, I would suggest having separate classes that imply the structure instead.

Reply:
you mean like we have models in parser-js, right?

This is really a good idea which could help us to extend the scope of this lib. Would love to implement it, but this isn't in the scope of the timeline. I would try to build lib as fast as possible so that I could implement these functionalities too at the end.